src/overview.md

```diff
@@ -0,0 +1,86 @@
+
+![Alt text](./JTAG_example.svg)
+
+[JTAG]: https://en.wikipedia.org/wiki/JTAG
+
+### SWD
+
+[SWD] is quite similar to JTAG, as it uses the same protocol but it is limited to two electrical signals and cannot connect multiple chips.
+While JTAG is vastly more versatile, SWD's strength is in its simplicity as many systems only has one micro controller to program and in these cases the reduction of signals while still having full debugability is a major win.
+In SWD the following signals are available:
+
+* SWDIO (Serial Wire Debug Input / Output)
+* SWCLK (Serial Wire Clock)
+
+In the following image you can see a common communication example with SWD communicating to a single micro controller:
+
+![Alt text](./SWD_example.svg)
+
+[SWD]: https://en.wikipedia.org/wiki/JTAG#Serial_Wire_Debug
+
+### Which one to use?
+
+When designing a system, we should select the interface which is most convenient for the system, and there is a simple criteria to start with: Does the systems need to program and debug more than one target? If yes, JTAG is a good choice, else SWD with its minimal amount of signals is a good choice.
+
+
+## Common debugger platforms
+
+### [OpenOCD]
+
+OpenOCD (Open On-Chip Debugger) is a very popular piece of software which provides an interface for a debugger software, such as GDB, to connect to in order to provide in-system programming and boundary-scan testing for embedded devices.
+However, OpenOCD is no hardware which is connected to, for example, a micro controller it needs this extra hardware known as a debug adapter to translate the debug commands into electrical signals which the system can understand.
+And as we talked about JTAG and SWD earlier, OpenOCD supports both of these signaling schemes through supported debug adapters which are commonly connected to the PC via USB and can be found from very cheap on for example eBay, or built into evaluation boards from micro controller manufacturers, to very expensive, special purpose, adapters.
+
+[OpenOCD]: http://openocd.org/
+
+### [Segger JLink]
+
+**I know nothing about JLink, Ryan, please add here :)**
+
+[Segger JLink]: https://www.segger.com/products/debug-probes/j-link/
+
+### [Black Magic Probe]
+
+The BMP (Black Magic Probe) is different from OpenOCD, where it is hardware, commonly the same as a debug adapter, which runs an GDB server over serial inside it, so there is no need to install OpenOCD - in a sense, it is OpenOCD combined with a debug adapter in the same dongle.
+The BMP is open hardware and runs open source software inside it as well, plus using GDB with the BMP is no different to OpenOCD, down to a few commands, so now there is a plethora of adapters to use.
+
+[Black Magic Probe]: https://github.com/blacksphere/blackmagic/wiki
+
+### Debugger used in this book
+
+In coherence with the Discovery Book, this book uses the STM32F3Discovery platform for all examples, which has an STLink v2 debug adapter which we will use OpenOCD to talk to.
```

Comment:
I wonder whether we could do that common API docs thing which lets you select your tools (usually programming language) for each example?
ime texane/stlink is simpler than getting OpenOCD working, what we default to may need more discussion?

Reply:
I mainly chose OpenOCD so we can touch on ITM and SWO (texane does not have this support).
If we don't need it I can change :)